src/Controls/src/Core/Element/Element.cs

```diff
@@ -333,30 +333,39 @@ internal Element ParentOverride
 		}
 
 		WeakReference<Element> _realParent;
-		/// <summary>For internal use by .NET MAUI.</summary>
-		[EditorBrowsable(EditorBrowsableState.Never)]
-		public Element RealParent
+		Element GetRealParent(bool logWarningIfParentHasBeenCollected = true)
```

Comment:
Not sure if possible, but, if this method might be called from multiple threads, consider adding synchronization around the `_realParent` assignment.

`readonly object _realParentLock = new object();`

Reply:
I think storing _realParent in a local variable is sufficient to handle this scenario. @jsuarezruiz 
 
https://github.com/dotnet/maui/pull/30156#discussion_r2177256986